storage/transfermanager/downloader.go

```diff
@@ -784,3 +894,57 @@ func setUsageMetricHeader(ctx context.Context, method string) context.Context {
 	header := fmt.Sprintf("%s/%s", usageMetricKey, method)
 	return callctx.SetHeaders(ctx, xGoogHeaderKey, header)
 }
+
+type crc32cPiece struct {
+	sum    uint32 // crc32c checksum of the piece
+	length int64  // number of bytes in this piece
+}
+
+// joinCRC32C pieces together the initial checksum with the orderedChecksums
+// provided to calculate the checksum of the whole.
+func joinCRC32C(initialChecksum uint32, orderedChecksums []crc32cPiece) uint32 {
```

Comment:
It seems surprising that this isn't available in hash/crc32 or elsewhere in a standard library, but I assume you looked.